### PR TITLE
fix read access and add write too

### PR DESCRIPTION
**Pull request by NickAzureDevops**
### Jira link (if applicable)

fix read access to create custom role to authenticate to cosmosdb 

### Checklist

- [x] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: components/custom_roles/role_definitions.yaml [hunk 1]
```diff
@@ -13,7 +13,8 @@ roles:
     description: "Custom role for DocumentDB with specific permissions"
     permissions:
       actions:
-        - "Microsoft.DocumentDB/databaseAccounts/readMetadata"
+        - "Microsoft.DocumentDB/databaseAccounts/read"
+        - "Microsoft.DocumentDB/databaseAccounts/write"
```
Comment (reespozzi) [inline components/custom_roles/role_definitions.yaml:17]:
```suggestion
```
Don't think we ned this sorry, we have containerWrite for the data

**Review by reespozzi — APPROVED** [review]

**Review by timja — CHANGES_REQUESTED** [review]
Why is this needed? there should be build in roles that do what you need for this.

you need to assign the data plane role using the special role assignment API for cosmos:
https://github.com/hmcts/slack-help-bot/blob/main/components/infrastructure/ai-cosmos.tf#L52-L59

**Comment by NickAzureDevops** [comment]
> Why is this needed? there should be built in roles that do what you need for this.
> 
> you need to assign the data plane role using the special role assignment API for cosmos: https://github.com/hmcts/slack-help-bot/blob/main/components/infrastructure/ai-cosmos.tf#L52-L59

it is for getting Managed identity to authenticate to cosmosdb @timja

**Review by timja — CHANGES_REQUESTED** [review]
> > Why is this needed? there should be built in roles that do what you need for this.
> > you need to assign the data plane role using the special role assignment API for cosmos: [hmcts/slack-help-bot@`main`/components/infrastructure/ai-cosmos.tf#L52-L59](https://github.com/hmcts/slack-help-bot/blob/main/components/infrastructure/ai-cosmos.tf?rgh-link-date=2024-07-23T15%3A56%3A39Z#L52-L59)
> 
> it is for getting Managed identity to authenticate to cosmosdb @timja

Then this is not needed, please revert all changes to this repo the previous link contains a working example with managed identity

**Pull request closed without merging**